main/fastcgi.c

```diff
@@ -1469,6 +1474,7 @@ int fcgi_accept_request(fcgi_request *req)
 			return -1;
 		}
 		if (fcgi_read_request(req)) {
+			req->hook.on_read();
```

Comment:
> doesn't that mean that in the case of fd < 0, we will end up calling on_read() twice
That can be the case, yes. Current implementation of php-fpm uses this hook just to zero variables, so it's not a problem. But if you think this should fire only once I can dig into it...
> Similarly, should on_read be called before fcgi_read_request() or after?
Again, since it's just resetting variables it doesn't matter as far as I understand. But I don't know what the author of that code really wanted for this hook.